lightning/src/chain/channelmonitor.rs

```diff
@@ -1669,6 +1669,28 @@ impl<Signer: WriteableEcdsaChannelSigner> ChannelMonitor<Signer> {
 			current_height, &broadcaster, &fee_estimator, &logger,
 		);
 	}
+
+	/// Returns the descriptor for a relevant output (i.e., one that we can spend) within the
+	/// transaction if one exists and the transaction has at least [`ANTI_REORG_DELAY`]
+	/// confirmations.
+	///
+	/// Descriptors returned by this method are primarily exposed via [`Event::SpendableOutputs`]
+	/// once they are no longer under reorg risk. This method can serve as a way to retrieve these
+	/// descriptors at a later time, either for historical purposes, or to replay any
+	/// missed/unhandled descriptors.
+	///
+	/// `tx` is a transaction we'll scan the outputs of. If an output which can be spent by us is
+	/// found, a descriptor is returned. `confirmation_height` must be the height of the block in
+	/// which `tx` was included in.
+	pub fn get_spendable_output(&self, tx: &Transaction, confirmation_height: u32) -> Option<SpendableOutputDescriptor> {
```

Comment:
Do we care that people may pass in 0 for `confirmation_height` and try to spend things early? Seems like there could be more safeguards in place.

Reply:
That sounds good, I'll defer to you and Matt on how important safeguards like that are.